clients/cobol-lsp-vscode-extension/src/ZoweApi.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import { BasicProfileManager, IProfile, Session, RestClient, AbstractSession } from "@zowe/imperative";
+import * as os from "os";
+import * as path from "path";
+
+export interface ProfilesMap {
+    [key: string]: IProfile;
+}
+
+export class ZoweApi {
+    private profileManager = new BasicProfileManager({
+        profileRootDirectory: path.join(os.homedir(), ".zowe", "profiles"),
+        type: "zosmf",
+    });
+
+    public async listZOSMFProfiles(): Promise<ProfilesMap> {
+        const profiles: ProfilesMap = {};
+        for (const loadedProfile of await this.profileManager.loadAll()) {
+            if (loadedProfile.failNotFound) {
+                profiles[loadedProfile.name] = loadedProfile.profile;
+            }
+        }
+        return profiles;
+    }
+
+    public getDefaultProfileName() {
+        return this.profileManager.getDefaultProfileName();
+    }
+
+    public async fetchMember(dataset: string, member: string, profileName: string): Promise<string> {
+        const session: Session = await this.createSession(profileName);
+
+        // default should be https
+        const rpath = `/zosmf/restfiles/ds/${dataset}(${member})`;
+        return await RestClient.getExpectString(session, rpath, [{
+            "Content-Type": "application/json", "X-CSRF-ZOSMF-HEADER": "" }]);
+    }
+
+    public async listMembers(dataset: string, profileName: string): Promise<string[]> {
+        // default should be https
+        const session: Session = await this.createSession(profileName);
+        const rpath = `/zosmf/restfiles/ds/${dataset}/member`;
```

Comment:
we are depending a lot on this hardcoded parts from Zowe

Reply:
It's not Zowe, it's zOSMF. Can you describe more, what problem do you see here?

Reply:
it is not a problem, the thing is that we hardcode the path and if they change something we are dependent on them

Reply:
I agree I would prefer to have these values in some resource file